shell/platform/darwin/macos/framework/Source/FlutterMenuPluginTest.mm

```diff
@@ -0,0 +1,170 @@
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterDartProject_Internal.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterMenuPlugin.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterMenuPlugin_Internal.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterTextInputSemanticsObject.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterViewController_Internal.h"
+#include "gtest/gtest.h"
+
+#import <OCMock/OCMock.h>
+#import "flutter/testing/testing.h"
+
+@interface FlutterMenuPluginTestObjc : NSObject
+- (bool)testSetMenu;
+@end
+
+@implementation FlutterMenuPluginTestObjc
+
+- (bool)testSetMenu {
+  // Build a simulation of the default main menu.
+  NSMenu* mainMenu = [[NSMenu alloc] init];
+  NSMenuItem* appNameMenu = [[NSMenuItem alloc] initWithTitle:@"APP_NAME"
+                                                       action:nil
+                                                keyEquivalent:@""];
+  NSMenu* submenu = [[NSMenu alloc] initWithTitle:@"Prexisting APP_NAME menu"];
+  [submenu addItem:[[NSMenuItem alloc] initWithTitle:@"About APP_NAME"
+                                              action:nil
+                                       keyEquivalent:@""]];
+  appNameMenu.submenu = submenu;
+  [mainMenu addItem:appNameMenu];
+  [NSApp setMainMenu:mainMenu];
+
+  id<FlutterPluginRegistrar> pluginRegistrarMock =
+      OCMProtocolMock(@protocol(FlutterPluginRegistrar));
+  __block FlutterMethodChannel* pluginChannel;
+  __block FlutterMenuPlugin* plugin;
+  id binaryMessengerMock = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  OCMStub([pluginRegistrarMock messenger]).andReturn(binaryMessengerMock);
+  OCMStub(  // NOLINT(google-objc-avoid-throwing-exception)
+      [pluginRegistrarMock addMethodCallDelegate:[OCMArg any] channel:[OCMArg any]])
+      .andDo(^(NSInvocation* invocation) {
+        id<FlutterPlugin> delegate;
+        FlutterMethodChannel* channel;
+        [invocation getArgument:&delegate atIndex:2];
+        [invocation getArgument:&channel atIndex:3];
+        pluginChannel = channel;
+        plugin = delegate;
+        [channel setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
+          [delegate handleMethodCall:call result:result];
+        }];
+      });
+  [FlutterMenuPlugin registerWithRegistrar:pluginRegistrarMock];
+
+  NSDictionary* testMenus = @{
+    @"0" : @[
+      @{
+        @"id" : [NSNumber numberWithInt:1],
+        @"label" : @"APP_NAME",
+        @"enabled" : @(YES),
+        @"children" : @[
+          @{
+            @"id" : [NSNumber numberWithInt:3],
+            @"platformProvidedMenu" : @(1),  // Quit
```

Comment:
```objc
#include "flutter/shell/platform/common/platform_provided_menu.h"

... snip ...

... @(static_cast<int>(flutter::PlatformProvidedMenu::kQuit)),
```

Reply:
What's the rule for `#include` vs `#import`?  If it's straight C++, use `#include`, and if it's mixed or ObjC use `#import`? Isn't it the case that we can just always use `#import`?